Add partitioning column configuration setting

johanhenriksson opened the pull request:
#### Description

Adds a configuration setting that allows the user to specify which column is used for partitioning. The column must be a unix millisecond timestamp.

This implementation seems to work for my use case, what do you think? @gliter 

Resolves #18 

##### PR Checklist
- [x] Tests added
- [x] [Changelog](CHANGELOG.md) updated

gliter commented on src/test/java/com/getindata/kafka/connect/iceberg/sink/TestIcebergUtil.java:
```diff
@@ -57,7 +57,7 @@ public void testUnwrapJsonRecord() throws IOException {
                 MAPPER.readTree(unwrapWithSchema).get("schema"), null);
         Schema schema = e.icebergSchema();
-        GenericRecord record = e.asIcebergRecord(schema);
+        GenericRecord record = e.asIcebergRecord(schema, "__source_ts_ms");
```
could be extracted as constant

johanhenriksson replied:
unsure where to put it, would be best to grab the default configuration setting. i guess it doesn't make much of a difference for the tests anyway :)

gliter replied:
I mean in that class as private static final to avoid having literal three times.

gliter commented:
Look good. Could you add some test case?

gliter commented:
Please also update changelog and readme.

johanhenriksson commented:
Added a simple test to ensure `IcebergChangeEvent` works as intended. Not sure how to properly test that the actual configuration setting is used, though. Also updated readme & changelog

johanhenriksson commented:
I will double check that everything works the way i want it tomorrow, don't merge until I mark it as ready :)

gliter commented on README.md:
```diff
@@ -27,7 +27,7 @@ mvn clean package
 | iceberg.*               |         |                | All properties with this prefix will be passed to Iceberg Catalog implementation                                                       |
 | iceberg.table-default.* |         |                | Iceberg specific table settings can be changed with this prefix, e.g. 'iceberg.table-default.write.format.default' can be set to 'orc' |
-
+| iceberg.partition       | String  | __source_ts_ms | Column used for partitioning. Must be a UTC unix millisecond timestamp.                                                                |
```
I have second thoughts about the name. Maybe `iceberg.partition.column` so we have some other partition related config in the future.

gliter commented:
Yep. That would be definitely an improvement.

gliter commented:
Not really. Maybe you could try to put some logging/breakpoint here:
https://github.com/getindata/kafka-connect-iceberg-sink/blob/develop/src/main/java/com/getindata/kafka/connect/iceberg/sink/IcebergUtil.java#L37
to see when tables are created

johanhenriksson commented:
Unfortunately I dont really have a local environment where I can easily debug :/ Weird thing is, there is already logging around the table creation, so I know it's not actually happening concurrently.

gliter commented:
Oh, i thought it happens during some automated tests. 
There is a system test with test containers, you can fiddle with it to check if you can replicate it.
https://github.com/getindata/kafka-connect-iceberg-sink/blob/develop/src/test/java/com/getindata/kafka/connect/iceberg/sink/IcebergSinkSystemTest.java

johanhenriksson commented:
The problems were unrelated to the connector. 

The latest commit introduces two settings:
 - `iceberg.partition.timestamp` decides which field is used as the timestamp source. Defaults to `__source_ms_ts`
 - `iceberg.partition.column` is the name of the output `timestamptz` column

If `iceberg.partition.timestamp` is empty, `iceberg.partition.column` is expected to already exist and be of the correct type. Not sure if this could actually happen with the current setup?

If they are set to the same field, that field is expected to be an integer timestamp, and its values will be converted to timestamptz and stored in a column with the same name, overwriting the integers.

Sounds good?
